praw/models/reddit/rules.py

```diff
@@ -0,0 +1,419 @@
+    ``"NAME"``:
+
+    .. code-block:: python
+
+        subreddit = reddit.subreddit("NAME")
+        rules = list(subreddit.rules)
+        new_rules = rules[3:4] + rules[1:3] + rules[0:1] + rules[4:]
+        # Alternate: [rules[3]] + rules[1:3] + [rules[0]] + rules[4:]
+        new_rule_list = subreddit.rules.mod.reorder(new_rules)
+
+    """
+
+    def __init__(self, subreddit_rules: SubredditRules):
+        """Instantize the SubredditRulesModeration class."""
+        self.subreddit_rules = subreddit_rules
+
+    def add(
+        self,
+        short_name: str,
+        kind: str,
+        description: str = "",
+        violation_reason: Optional[str] = None,
+    ) -> Rule:
+        """Add a removal reason to this subreddit.
+
+        :param short_name: The name of the rule.
+        :param kind: The kind of item that the rule applies to. One of
+            ``"link"``, ``"comment"``, or ``"all"``.
+        :param description: The description for the rule. Optional.
+        :param violation_reason: The reason that is shown on the report menu.
+            If a violation reason is not specified, the short name will be used
+            as the violation reason.
+        :returns: The Rule added.
+
+        To add rule ``"No spam"`` to the subreddit ``"NAME"`` try:
+
+        .. code-block:: python
+
+           reddit.subreddit('NAME').rules.mod.add(
+               "No spam",
+               "all",
+               description="Do not spam. Spam bad")
+
+        """
+        data = {
+            "r": str(self.subreddit_rules.subreddit),
+            "description": description,
+            "kind": kind,
+            "short_name": short_name,
+            "violation_reason": short_name
+            if violation_reason is None
+            else violation_reason,
+        }
+        new_rule = self.subreddit_rules._reddit.post(
+            API_PATH["add_subreddit_rule"], data=data
+        )[0]
+        new_rule.subreddit = self.subreddit_rules.subreddit
+        return new_rule
+
+    def reorder(self, rule_list: List[Rule]) -> List[Rule]:
+        """Reorder the rules of a subreddit.
+
+        :param rule_list: The list of rules, in the wanted order. Each index of
+            the list indicates the position of the rule.
+        :returns: A list containing the rules in the specified order.
+
+        For example, to move the fourth rule to the first position, and then to
+        move the prior first rule to where the third rule originally was in the
+        subreddit ``"NAME"``:
+
+        .. code-block:: python
+
+            subreddit = reddit.subreddit("NAME")
+            rules = list(subreddit.rules)
+            new_rules = rules[3:4] + rules[1:3] + rules[0:1] + rules[4:]
+            # Alternate: [rules[3]] + rules[1:3] + [rules[0]] + rules[4:]
+            new_rule_list = subreddit.rules.mod.reorder(new_rules)
+
+        """
+        order_string = quote(
+            ",".join([str(item) for item in rule_list]), safe=","
```

Comment:
Is `item` ever going to be something that's not a string? If so, let's drop the `str` wrapper.

Reply:
Actually, in this case, I am assuming that the list will be instances of the Rule class, so I am assuming that 100% of the use cases are going to be non-strings.

Reply:
I see. In that case, is there an attribute we can call directly, rather than relying on `str`?

Reply:
Since the string attribute is the short_name, we could call that.